arangod/Replication/Syncer.cpp

```diff
@@ -674,6 +675,98 @@ Result Syncer::dropIndex(arangodb::velocypack::Slice const& slice) {
 
   return r;
 }
+  
+/// @brief creates a view, based on the VelocyPack provided
+Result Syncer::createView(TRI_vocbase_t& vocbase,
+                          arangodb::velocypack::Slice const& slice) {
+  if (!slice.isObject()) {
+    return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
+                  "collection slice is no object");
+  }
+  
+  VPackSlice nameSlice = slice.get(StaticStrings::DataSourceName);
+  if (!nameSlice.isString() || nameSlice.getStringLength() == 0) {
+    return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
+                  "no name specified for view");
+  }
+  VPackSlice guidSlice = slice.get("globallyUniqueId");
+  if (!guidSlice.isString() || guidSlice.getStringLength() == 0) {
+    return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
+                  "no guid specified for view");
+  }
+  VPackSlice typeSlice = slice.get(StaticStrings::DataSourceType);
+  if (!typeSlice.isString() || typeSlice.getStringLength() == 0) {
+    return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
+                  "no type specified for view");
+  }
+  
+  auto view = vocbase.lookupView(guidSlice.copyString());
+  if (view) {
+    return {}; // view already exists. TODO: compare attributes
+  }
+  // resolve name conflicts by deleting existing
+  view = vocbase.lookupView(nameSlice.copyString());
+  if (view) {
+    Result res = vocbase.dropView(view->id(), /*dropSytem*/false);
```

Comment:
should the code really drop the view with the same name instead of returning duplicate-name-error?

Reply:
I think that is the behaviour for collections

Reply:
quite dangerous if user-triggerable